notification-service/src/main/java/org/arpha/notificationservice/domain/Notification.java

```diff
@@ -0,0 +1,48 @@
+package org.arpha.notificationservice.domain;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.EntityListeners;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.NoArgsConstructor;
+import org.arpha.domain.type.notification.NotificationStatus;
+import org.arpha.domain.type.notification.NotificationType;
+import org.arpha.notificationservice.domain.type.DeliveryType;
+import org.hibernate.annotations.CreationTimestamp;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import java.time.OffsetDateTime;
+
+@Builder
+@Entity
+@Table(name = "notifications")
+@AllArgsConstructor
+@NoArgsConstructor
+@EntityListeners(AuditingEntityListener.class)
+public class Notification {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Enumerated(EnumType.STRING)
+    private DeliveryType deliveryType;
+
+    @Enumerated(EnumType.STRING)
+    private NotificationType notificationType;
+
+    @Enumerated(EnumType.STRING)
+    private NotificationStatus status;
+
+    private String recipient;
+
+    @CreationTimestamp
+    private OffsetDateTime createdAt;
```

Comment:
Why do we use here OffsetDateRine? Can't we use localDateTime?

Reply:
I am using OffsetDateTime because it also stores time zone information. LocalDateTime on the other hand does not store any time zone information, thats why I am using OffsetDateTime in conjunction with (timestamp with time zone db datatype)